examples/complete/variables.tf

```diff
@@ -131,6 +131,22 @@ variable "zonal_shift_config" {
   default     = null
 }
 
+variable "cluster_compute_config" {
+  type = object({
+    enabled       = optional(bool, false)
+    node_pools    = optional(list(string), [])
+    node_role_arn = optional(string, null)
+  })
+  description = <<-EOT
+    Configuration block with compute configuration for EKS Auto Mode
+
+    enabled: Request to enable or disable the compute capability on your EKS Auto Mode cluster. If the compute capability is enabled, EKS Auto Mode will create and delete EC2 Managed Instances in your Amazon Web Services account.
+    node_pools: Optional configuration for node pools that defines the compute resources for your EKS Auto Mode cluster. Valid options are general-purpose and system.
+    node_role_arn: Optional ARN of the IAM Role EKS will assign to EC2 Managed Instances in your EKS Auto Mode cluster.
```

Comment:
Are there any gotchas with the default IAM role EKS will use, e.g [edge cases](https://github.com/cloudposse/terraform-aws-eks-cluster/blob/6521512dd3d554acabfdd6840b47c0acd5b2ec89/iam.tf#L44-L46)? If so, then we should probably have an option to create and use the role we create when we use the old/existing node pool. In fact, that should probably be an option in any case.